media/js/firefox/onboarding/user-actions.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function(Mozilla, $) {
+    'use strict';
+
+    var client = Mozilla.Client;
+
+    // *************
+    // pin it/search variation
+    // *************
+    var $trySearch = $('#search-now');
+    var $searchPointer = $('#search-pointer');
+
+    // as this is for firstrun, search should almost certainly be available
+    // best to double check though
+    function shouldPromoteSearch() {
+        var availableTargets = new Promise(function(resolve, reject) {
+            Mozilla.UITour.getConfiguration('availableTargets', function(config) {
+                if (config && config.targets) {
+                    resolve(config.targets);
+                } else {
+                    reject('UITour: targets property not found.');
+                }
+            });
+        });
+
+        return new Promise(function(resolve, reject) {
+            Promise.all([availableTargets]).then(function(results) {
+                resolve(results[0] && results[0].indexOf('search') > -1);
+            }, function(reason) {
+                reject(reason);
+            });
+        });
+    }
+
+    function openSearchUI(e) {
+        e.preventDefault();
+        Mozilla.UITour.openSearchPanel(function() {
+            Mozilla.UITour.setSearchTerm('What is the weather?');
+            $trySearch.addClass('hidden');
```

Comment:
I tried to push back against this in the Yahoo Search page a bit (and lost out), but I think this is a bit of a UX anti-pattern. First time I clicked the button I accidentally closed the search menu again. I do know where the menu is, but my first reaction was to try and click the button again.